slack/resource_conversation.go

```diff
@@ -246,12 +264,19 @@ func resourceSlackConversationDelete(ctx context.Context, d *schema.ResourceData
 	client := m.(*slack.Client)
 
 	id := d.Id()
-	err := archiveConversationWithContext(ctx, client, id)
-	if err != nil {
-		if err.Error() == "channel_not_found" {
-			return diags
+	action := d.Get("action_on_destroy").(string)
+	switch action {
```

Comment:
even though you are using a validate function in the schema I would prefer if a `default` clause was included here as well, just to be more defensive.

```go
switch action {
    default: 
         return diag.Errorf("unknown action_on_destroy value. Valid values are %v", conversationActionValidValues)
}
```

This helps for example if you added a new valid value that the schema validator accepts but it is not added to the switch clause, or misspelled. 

You would have `conversationActionValidValues` as a var:

```go
var (
	conversationActionValidValues = []string{
		conversationActionOnDestroyNone,
		conversationActionOnDestroyArchive,
	}
	validateConversationActionOnDestroyValue = validation.StringInSlice(conversationActionValidValues, false)
)
```

Reply:
fixed